text/0171-cloudfront-redesign.md

```diff
@@ -0,0 +1,484 @@
+documents.
+
+```ts
+import * as cloudfront from '@aws-cdk/aws-cloudfront';
+
+// Creates a distribution for a S3 bucket.
+const myBucket = new s3.Bucket(...);
+new cloudfront.Distribution(this, 'myDist', {
+  origin: cloudfront.Origin.fromBucket(myBucket),
+});
+
+// Creates a distribution for a S3 bucket that has been configured for website hosting.
+const myWebsiteBucket = new s3.Bucket(...);
+new cloudfront.Distribution(this, 'myDist', {
+  origin: cloudfront.Origin.fromWebsiteBucket(myBucket),
+});
+```
+
+Both of the S3 Origin options will automatically create an origin access identity and grant it access to the underlying bucket. This can be used in
+conjunction with a bucket that is not public to require that your users access your content using CloudFront URLs and not S3 URLs directly.
+
+### From an HTTP endpoint
+
+Origins can also be created from other resources (e.g., load balancers, API gateways), or from any accessible HTTP server.
+
+```ts
+// Creates a distribution for an application load balancer.
+const myLoadBalancer = new elbv2.ApplicationLoadBalancer(...);
+new cloudfront.Distribution(this, 'myDist', {
+  origin: cloudfront.Origin.fromLoadBalancerV2(myLoadBalancer),
+});
+
+// Creates a distribution for an HTTP server.
+new cloudfront.Distribution(this, 'myDist', {
+  origin: cloudfront.Origin.fromHTTPServer({
+    domainName: 'www.example.com',
+    protocolPolicy: OriginProtocolPolicy.HTTPS_ONLY,
+  })
+});
+```
+
+## Domain Names and Certificates
+
+When you create a distribution, CloudFront returns a domain name for the distribution, for example: `d111111abcdef8.cloudfront.net`. If you want to
+use your own domain name, such as `www.example.com`, you can add an alternate domain name to your distribution.
+
+```ts
+new cloudfront.Distribution(this, 'myDist', {
+  origin: cloudfront.Origin.fromBucket(myBucket),
+  aliases: ['www.example.com'],
+})
+```
+
+CloudFront distributions use a default certificate (`*.cloudfront.net`) to support HTTPS by default. If you want to support HTTPS with your own domain
+name, you must associate a certificate with your distribution that contains your domain name. The certificate must be present in the AWS Certificate
+Manager (ACM) service in the US East (N. Virginia) region; the certificate may either be created by ACM, or created elsewhere and imported into ACM.
+
+```ts
+const myCertificate = new certmgr.DnsValidatedCertificate(this, 'mySiteCert', {
+  domainName: 'www.example.com',
+  hostedZone,
+});
+new cloudfront.Distribution(this, 'myDist', {
+  origin: cloudfront.Origin.fromBucket(myBucket),
+  certificate: myCertificate,
+});
+```
+
+Note that in the above example the aliases are inferred from the certificate and do not need to be explicitly provided.
+
+## Caching Behaviors
+
+Each distribution has a default cache behavior which applies to all requests to that distribution; additional cache behaviors may be specified for a
```

Comment:
Is this about caching or not? If not, maybe just calling it "behaviors" makes more sense (maybe indicate that upstream uses "cache behaviors" but that this no longer covers the cases).